src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc

```diff
@@ -136,13 +136,15 @@ dword_result_t NtAllocateVirtualMemory(lpdword_t base_addr_ptr,
   uint32_t protect = FromXdkProtectFlags(protect_bits);
   uint32_t address = 0;
   BaseHeap* heap;
+  HeapAllocationInfo* prev_alloc_info = new HeapAllocationInfo();
```

Comment:
You are leaking memory.

`HeapAllocationInfo prev_alloc_info = {};`

`heap->QueryRegionInfo(adjusted_base, &prev_alloc_info);`

etc

Reply:
I'd say it's probably best as something like:

```c
HeapAllocationInfo prev_alloc_info = {};
bool was_commit = heap->QueryRegionInfo(adjusted_base, &prev_alloc_info) && (prev_alloc_info.kMemoryAllocationCommit & kMemoryAllocationCommit) != 0;
```

Then later use `if (!was_commit)`.

Reply:
I disagree with second post. 
I would need to make ``bool was_commit`` set it from default to false and then inside branches aka:
 ```if (adjusted_base != 0) {```
get that info and store it to that boolean, but at this point we have it already in prev_alloc_info, so imo usage of ``bool was_commit`` is quite unnecessary.

I will just add note that should resolve eventual confusion

Reply:
The point here is to never used an uninitialized `prev_alloc_info`; even if we initialize it to zero, there's no true guarantee as to what it is in `prev_alloc_info` if `QueryRegionInfo` returns `false`.

Reply:
Resolved